openapi/src/main/java/io/helidon/openapi/OpenAPISupport.java

```
@@ -445,165 +323,208 @@ public static Builder builder() {
-         * @param path path to which the server corresponds
-         * @param pathServer name of the server to add for this path
-         * @return updated builder instance
-         */
-        public Builder addPathServer(String path, String pathServer) {
-            Objects.requireNonNull(path, "path must be non-null");
-            Objects.requireNonNull(pathServer, "pathServer must be non-null");
-            apiConfigBuilder.addPathServer(path, pathServer);
-            return this;
+        private OpenApiStaticFile getExplicitStaticFile() {
+            Path path = Paths.get(staticFilePath.get());
+            final String specifiedFileType = typeFromPath(path);
+            final OpenAPIMediaTypes specifiedMediaType = OpenAPIMediaTypes.byFileType(specifiedFileType);
+
+            if (specifiedMediaType == null) {
+                throw new IllegalArgumentException("OpenAPI file path "
+                        + path.toAbsolutePath().toString()
+                        + " is not one of recognized types: "
+                        + OpenAPIMediaTypes.recognizedFileTypes());
+            }
+            final InputStream is;
+            try {
+                is = Files.newInputStream(path);
+            } catch (IOException ex) {
+                throw new IllegalArgumentException("OpenAPI file "
+                        + path.toAbsolutePath().toString()
+                        + " was specified but was not found");
+            }
+
+            try {
+                LOGGER.log(Level.FINE,
+                        () -> String.format(
+                                OPENAPI_EXPLICIT_STATIC_FILE_LOG_MESSAGE_FORMAT,
+                                path.toAbsolutePath().toString()));
+                return new OpenApiStaticFile(is, specifiedMediaType.format());
+            } catch (Throwable th) {
+                try {
+                    is.close();
+                } catch (IOException ex) {
+                    LOGGER.log(Level.FINE,
```

Consider adding a suppressed throwable instead of logging.

---

Yes, better. Fixed.